config/README.md

```diff
@@ -34,13 +35,18 @@ type NewRelicConfig struct {
 
 func main() {
 	var c Config
+	cmd := cobra.Command{}
+	cmd.Flags().Int("db-port", 5432, "set db port")
```

Comment:
Looking at this usage example, can we automate setting the flag definitions?

Since we support setting the default values in the struct, there can be a mismatch in the value set by flag vs that in the struct. Basically we won't have a single source of truth.

So we have 2 options here -
- Support defining flags with this config package (Not sure how complex this would be)
- Remove support for default values from struct and only use default values from flags. (Easy, but this option has an issue that folks not using flags will need to figure out their own way to set defaults)

Reply:
@rohilsurana @deryrahman Any final decision on this?

Reply:
not yet @ravisuhag , deferring the discussion as I haven't spiked the approach yet

I strongly suggest not to remove the support default values from struct